File: Programming Basics/02-Conditional Statements - Exercise/toyShop.js
function toyShop(input) {

    const PUZZLE = 2.60;
    const DOLL = 3.00;
    const BEAR = 4.10;
    const MINION = 8.20;
    const TRUCK = 2.00;

    let priceForExcursion = Number(input[0]);
    let numberOfPuzzle = Number(input[1]);
    let numberOfDoll = Number(input[2]);
    let numberOfBear = Number(input[3]);
    let numberOfMinion = Number(input[4]);
    let numberOFTruck = Number(input[5]);

    let rent = 0.10;

    let raisedMoney = PUZZLE * numberOfPuzzle + DOLL * numberOfDoll + BEAR * numberOfBear + MINION * numberOfMinion + TRUCK * numberOFTruck;

    if(numberOfPuzzle + numberOfDoll + numberOfBear + numberOfMinion + numberOFTruck >= 50) {
        raisedMoney = raisedMoney - (raisedMoney * 0.25);
    }
    
    raisedMoney = raisedMoney - (raisedMoney *rent);

    let diffrence = Math.abs(raisedMoney - priceForExcursion);

    if(raisedMoney >= priceForExcursion) {
        console.log(`Yes! ${diffrence.toFixed(2)} lv left.`);
    } else {
        console.log(`Not enough money! ${diffrence.toFixed(2)} lv needed.`);
    }

}

toyShop(["40.8",
"20",
"25",
"30",
"50",
"10"])

File: Programming Basics Java/src/ConditionalStatementsExercise/ToyShop.java
package ConditionalStatementsExercise;

import java.util.Scanner;

public class ToyShop {
    public static void main(String[] arg) {
        Scanner scan = new Scanner(System.in);

        double excursionPrice = Double.parseDouble(scan.nextLine());

        int numberOfPuzzle = Integer.parseInt(scan.nextLine());
        int numberOfDolls = Integer.parseInt(scan.nextLine());
        int numberOfBears = Integer.parseInt(scan.nextLine());
        int numberOfMinion = Integer.parseInt(scan.nextLine());
        int numberOfTrucks = Integer.parseInt(scan.nextLine());

        double priceOfPuzzle = numberOfPuzzle * 2.60;
        double priceOfDolls = numberOfDolls * 3.00;
        double priceOfBears = numberOfBears * 4.10;
        double priceOfMinion = numberOfMinion * 8.20;
        double priceOfTrucks = numberOfTrucks * 2.00;

        double totalPrice = priceOfPuzzle + priceOfDolls + priceOfBears + priceOfMinion + priceOfTrucks;
        int totalToys = numberOfPuzzle + numberOfDolls + numberOfBears + numberOfMinion + numberOfTrucks;

        if (totalToys >= 50) {
            totalPrice *= 0.75;
        }

        totalPrice *= 0.90;

        double profit = Math.abs(totalPrice - excursionPrice);

        if (totalPrice >= excursionPrice) {
            System.out.printf("Yes! %.2f lv left.", profit);
        } else {
            System.out.printf("Not enough money! %.2f lv needed.", profit);
        }
    }
}
